test/distributed/_tools/test_fsdp2_mem_tracker.py

```diff
@@ -0,0 +1,238 @@
+# Owner(s): ["module: unknown"]
+import functools
+from typing import Union
+
+import torch
+import torch.nn as nn
+from torch.distributed._composable import checkpoint
+from torch.distributed._composable.fsdp import (
+    CPUOffloadPolicy,
+    fully_shard,
+    MixedPrecisionPolicy,
+    OffloadPolicy,
+)
+from torch.distributed._tensor import init_device_mesh
+from torch.distributed._tools.fsdp2_mem_tracker import FSDPMemTracker
+from torch.distributed.algorithms._checkpoint.checkpoint_wrapper import (
+    apply_activation_checkpointing,
+    CheckpointWrapper,
+)
+
+from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
+from torch.testing._internal.common_fsdp import FSDPTest, MLP
+from torch.testing._internal.common_utils import run_tests
+from torch.testing._internal.distributed._tensor.common_dtensor import (
+    ModelArgs,
+    Transformer,
+    TransformerBlock,
+)
+
+c10d_ops = torch.ops.c10d
+funcol = torch.ops.c10d_functional
+
+
+class TestTrackerFullyShard1DTrainingCore(FSDPTest):
+    @property
+    def world_size(self) -> int:
```

Comment:
I think if you do not add this to run in a multi-GPU test, then it will never have 4 GPUs (only 2).

See
https://github.com/pytorch/pytorch/blob/9d1b65b569a304032776375c24aa6babbf576c30/.ci/pytorch/test.sh#L325-L326
for where we sneak in some tests to run with Inductor tests that have 4 GPUs.

Reply:
For now, I will skip adding anything to CI. Once people use the tracker and we harden it, then we can add more tests.